tests/e2e/Services/Projects/ProjectsConsoleClientTest.php

```diff
@@ -3461,4 +3462,498 @@ public function testDeleteProject(): array
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project2['body']['$id'],
+            'x-appwrite-key' => $key2['body']['secret']
+        ]);
+
+        $this->assertEquals(404, $collectionProject1WithProject2Key['headers']['status-code']);
+
+        // Attempt to read cross-tenant resources
+        $collectionProject3WithProject1Key = $this->client->call(Client::METHOD_GET, '/databases/' . $database3['body']['$id'] . '/collections/' . $collection3['body']['$id'], [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project1['body']['$id'],
+            'x-appwrite-key' => $key1['body']['secret']
+        ]);
+
+        $this->assertEquals(404, $collectionProject3WithProject1Key['headers']['status-code']);
+
+        $collectionProject1WithProject3Key = $this->client->call(Client::METHOD_GET, '/databases/' . $database1['body']['$id'] . '/collections/' . $collection1['body']['$id'], [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project3['body']['$id'],
+            'x-appwrite-key' => $key3['body']['secret']
+        ]);
+
+        $this->assertEquals(404, $collectionProject1WithProject3Key['headers']['status-code']);
+
+        // Assert that shared project resources can have the same ID as they're unique on tenant + ID not just ID
+        $collection5 = $this->client->call(Client::METHOD_POST, '/databases/' . $database2['body']['$id'] . '/collections', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project2['body']['$id'],
+            'x-appwrite-key' => $key2['body']['secret']
+        ], [
+            'databaseId' => $database2['body']['$id'],
+            'collectionId' => $collection4['body']['$id'],
+            'name' => 'Amazing Collection',
+        ]);
+
+        // Assert that users across projects on shared tables can have the same email as they're unique on tenant + email not just email
+        $user1 = $this->client->call(Client::METHOD_POST, '/users', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project2['body']['$id'],
+            'x-appwrite-key' => $key2['body']['secret']
+        ], [
+            'userId' => 'user',
+            'email' => 'test@appwrite.io',
+            'password' => 'password',
+            'name' => 'Test User',
+        ]);
+
+        $this->assertEquals(201, $user1['headers']['status-code']);
+
+        $user2 = $this->client->call(Client::METHOD_POST, '/users', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $project4['body']['$id'],
+            'x-appwrite-key' => $key4['body']['secret']
+        ], [
+            'userId' => 'user',
+            'email' => 'test@appwrite.io',
+            'password' => 'password',
+            'name' => 'Test User',
+        ]);
+
```

Comment:
Aren't there a lot more resources we should check for? Or is it sufficient to just check for the users table and database table?

Reply:
It's enough to check a per-project collection and a shared collection, since those are the only types of collections, any issues would present